src/platforms/windows/daemon/windowsfirewall.cpp

```diff
@@ -0,0 +1,874 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "windowsfirewall.h"
+#include "logger.h"
+#include "leakdetector.h"
+#include "../windowscommons.h"
+#include "../../daemon/interfaceconfig.h"
+
+#include <QApplication>
+#include <QObject>
+#include <QFileInfo>
+#include <QNetworkInterface>
+#include <QScopeGuard>
+#include <QHostAddress>
+#include <QtEndian>
+#include <windows.h>
+#include <fwpmu.h>
+#include <stdio.h>
+#include <comdef.h>
+#include <netfw.h>
+#include "winsock.h"
```

Comment:
<winsock.h> ?

Reply:
Definces proto numbers for wfp filters: i.e
`conds[0].conditionValue.uint8 = (IPPROTO_UDP);`